### PR TITLE
Enhancements

### PR DESCRIPTION
**Pull request by jesuscol96**
- Make configurable
- Allow data clearing

### CODE REVIEW

File: target_gsheet.py [hunk 1]
```diff
@@ -48,18 +46,15 @@ def get_credentials():
     Returns:
         Credentials, the obtained credential.
     """
-    home_dir = os.path.expanduser('~')
-    credential_dir = os.path.join(home_dir, '.credentials')
-    if not os.path.exists(credential_dir):
-        os.makedirs(credential_dir)
-    credential_path = os.path.join(credential_dir,
-                                   'sheets.googleapis.com-singer-target.json')
+    client_secret = config.get('clientSecret', 'client_secret.json')
```
Comment (brava-vinh) [inline target_gsheet.py]: should use `config.get('clientSecret', CLIENT_SECRET_FILE)` to avoid magic val

File: target_gsheet.py [hunk 2]
```diff
@@ -119,13 +122,15 @@ def flatten(d, parent_key='', sep='__'):
             items.append((new_key, str(v) if type(v) is list else v))
     return dict(items)
 
-def persist_lines(service, spreadsheet, lines):
+def persist_lines(service, spreadsheet, lines, clear_sheets=False):
```
Comment (brava-vinh) [inline target_gsheet.py]: we shouldn't pass clear_sheets but instead we should pass the whole config and in config.json we should have sync setup to replace or append.

Comment (brava-vinh) [reply target_gsheet.py:125]: we may not even need to pass the config object because it's already on the class?

File: target_gsheet.py [hunk 3]
```diff
@@ -140,12 +145,16 @@ def persist_lines(service, spreadsheet, lines):
             schema = schemas[msg.stream]
             validate(msg.record, schema)
             flattened_record = flatten(msg.record)
-            
+
             matching_sheet = [s for s in spreadsheet['sheets'] if s['properties']['title'] == msg.stream]
```
Comment (brava-vinh) [inline target_gsheet.py]: one thing I don't like about those singer target is they don't allow us to overwrite the sheet(the tab in a google sheet) and relying on title matching of stream only. can you add a `sheet_title` or sth like that in config.json so we can force it to the exact sheet we want?

File: target_gsheet.py (hunk 2 above)
Comment (jesuscol96) [reply target_gsheet.py]: Done

File: target_gsheet.py (hunk 3 above)
Comment (jesuscol96) [reply target_gsheet.py]: Good catch. Done

File: target_gsheet.py (hunk 1 above)
Comment (jesuscol96) [reply target_gsheet.py]: Fixed

**Comment by jesuscol96** [comment]
I also added batch loading. It was making one HTTP request per row of data. Pretty inefficient, API got max calls every time. Now it works nicely. Definitely, this was impossible to use without forking. @brava-vinh

**Review by brava-vinh — APPROVED** [review]

**Pull request merged**